src/librustc_mir/build/matches/simplify.rs

```diff
@@ -92,11 +92,19 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                 Err(match_pair)
             }
 
-            PatternKind::Range { .. } |
-            PatternKind::Slice { .. } => {
+            PatternKind::Range { .. } => {
                 Err(match_pair)
             }
 
+            PatternKind::Slice { ref prefix, ref slice, ref suffix } => {
+                if prefix.is_empty() && slice.is_some() && suffix.is_empty() {
+                    // irrefutable
+                    Ok(())
```

Comment:
Something seems missing here. Doesn't this produce a binding for the slice? If so, don't we have to extend `candidate.bindings`?

Maybe try an example that actually uses the binding? e.g.

```rust
let [foo..] = s;
println!("{:?}", foo);
```

Reply:
Thank you for you review! I tried the example, and it died with segmentation fault. So yes, we need to extend `candidate.bindings`. I will update this PR to do so.